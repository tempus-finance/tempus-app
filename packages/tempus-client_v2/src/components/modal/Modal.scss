.tc__modal-backdrop {
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 2;
  cursor: default;
}

.tc__modal {
  min-width: 340px;
  min-height: 75px;
  background-color: white;
  border: 1px solid #f3f7f9;
  box-sizing: border-box;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.08);
  border-radius: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
}

.tc__modal-header {
  height: 48px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.tc__modal-header-close {
  cursor: pointer;
}
