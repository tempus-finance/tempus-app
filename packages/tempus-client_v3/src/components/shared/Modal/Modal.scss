@mixin overlay {
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;
}

.tc__modal-backdrop {
  @include overlay;
  background-color: var(--modalBackdrop);
  z-index: 9998;
}

.tc__modal-container {
  @include overlay;
  display: flex;
  align-items: center;
  justify-content: center;
  pointer-events: none;
  z-index: 9999;
}

.tc__modal {
  position: relative;
  border: 1px solid var(--modalBorder);
  border-radius: 12px;
  background-color: var(--modalBackground);
  padding: 16px;
  pointer-events: all;
}

.tc__modal-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 16px;
}