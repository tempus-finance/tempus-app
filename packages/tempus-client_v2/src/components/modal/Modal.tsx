import { FC, useContext } from 'react';
import { LanguageContext } from '../../context/languageContext';
import getText from '../../localisation/getText';
import Words from '../../localisation/words';
import CloseIcon from '../icons/CloseIcon';
import Typography from '../typography/Typography';

import './Modal.scss';

interface ModalProps {
  open: boolean;
  title: Words;
  onClose: () => void;
}

const Modal: FC<ModalProps> = ({ onClose, open, title, children }) => {
  const { language } = useContext(LanguageContext);

  if (!open) {
    return null;
  }

  return (
    <div className="tc__modal-backdrop">
      <div className="tc__modal">
        <div className="tc__modal-header">
          {/* TODO - Update typography style once new fonts are merged into 2.0 branch */}
          <Typography variant="h4">{getText(title, language)}</Typography>
          <div className="tc__modal-header-close" onClick={onClose}>
            <CloseIcon />
          </div>
        </div>
        {children}
      </div>
    </div>
  );
};
export default Modal;
