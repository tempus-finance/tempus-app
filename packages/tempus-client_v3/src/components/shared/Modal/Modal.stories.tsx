import { ComponentStory, ComponentMeta } from '@storybook/react';
import { useCallback, useState } from 'react';
import Typography from '../Typography';

import Modal from './Modal';

export default {
  title: 'Modal',
  component: Modal,
  argTypes: {},
} as ComponentMeta<typeof Modal>;

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template: ComponentStory<typeof Modal> = () => {
  const [modalOpen, setModalOpen] = useState(false);

  const toggleModal = () => {
    setModalOpen(prevState => !prevState);
  };

  const onModalClose = useCallback(() => {
    setModalOpen(false);
  }, []);

  return (
    <div style={style}>
      <button type="button" onClick={toggleModal}>
        Click me!
      </button>

      <Modal open={modalOpen} onClose={onModalClose}>
        <Typography variant="body-primary">Modal content</Typography>
      </Modal>
    </div>
  );
};

export const Primary = Template.bind({});
