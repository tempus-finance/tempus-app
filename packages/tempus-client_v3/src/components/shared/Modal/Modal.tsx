import React, { FC, ReactNode, useCallback } from 'react';
import ButtonWrapper from '../ButtonWrapper';
import Icon from '../Icon';
import Typography from '../Typography';

import './Modal.scss';

type ModalVariant = 'plain' | 'styled';
type ModalSize = 'small' | 'large';

export interface ModalProps {
  open: boolean;
  onClose: () => void;
  title?: string;
  className?: string;
  header?: ReactNode;
}

export interface ModalStyleProps {
  variant?: ModalVariant;
  size?: ModalSize;
}

const Modal: FC<ModalProps & ModalStyleProps> = props => {
  const { title = '', open, onClose, className = '', variant = 'plain', size = 'small', header, children } = props;

  const onBackdropClick = (event: React.MouseEvent<HTMLDivElement>) => {
    event.stopPropagation();

    onClose();
  };

  const onCloseButtonClick = useCallback(() => {
    onClose();
  }, [onClose]);

  if (!open) {
    return null;
  }

  return (
    <>
      <div id="modal-backdrop" className="tc__modal-backdrop" onClick={onBackdropClick} />
      <div className={`tc__modal-container ${className}`}>
        <div className={`tc__modal tc__modal__${variant} tc__modal__size-${size}`}>
          <div className="tc__modal-header">
            <Typography variant="body-primary" weight="bold">
              {title}
            </Typography>
            <ButtonWrapper onClick={onCloseButtonClick}>
              <Icon variant="close" size={size === 'small' ? 'small' : 20} />
            </ButtonWrapper>
            {header && <div className="tc__modal-header-content">{header}</div>}
          </div>
          <div className="tc__modal__body">{children}</div>
        </div>
      </div>
    </>
  );
};
export default React.memo(Modal) as FC<ModalProps & ModalStyleProps>;
