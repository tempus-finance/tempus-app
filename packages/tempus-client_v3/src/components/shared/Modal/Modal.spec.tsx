import { fireEvent, render } from '@testing-library/react';
import Modal, { ModalProps, ModalStyleProps } from './Modal';

const mockOnCloseHandler = jest.fn();

const defaultProps: ModalProps = {
  open: true,
  onClose: mockOnCloseHandler,
};

const modalContentText = 'This is modal content.';

const subject = (props: ModalProps & ModalStyleProps) => render(<Modal {...props}>{modalContentText}</Modal>);

describe('Modal', () => {
  it('renders child elements when small modal is open', () => {
    const { container, getByText } = subject({ ...defaultProps, size: 'small' });

    const modalContent = getByText(modalContentText);

    expect(modalContent).not.toBeNull();

    expect(container).toMatchSnapshot();
  });

  it('renders child elements when large modal is open', () => {
    const { container, getByText } = subject({ ...defaultProps, size: 'large' });

    const modalContent = getByText(modalContentText);

    expect(modalContent).not.toBeNull();

    expect(container).toMatchSnapshot();
  });

  it('does not render child elements when modal is closed', () => {
    const { container, queryByText } = subject({ open: false, onClose: mockOnCloseHandler });

    const modalContent = queryByText(modalContentText);

    expect(modalContent).toBeNull();

    expect(container).toMatchSnapshot();
  });

  it('calls onClose when user clicks on backdrop', () => {
    const { container } = subject(defaultProps);

    const backdrop = container.querySelector('#modal-backdrop');
    expect(backdrop).not.toBeNull();

    if (backdrop) {
      fireEvent.click(backdrop);
    }

    expect(mockOnCloseHandler).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });

  it('calls onClose when user clicks on the close button', () => {
    const { getByRole } = subject(defaultProps);

    const closeButton = getByRole('button');
    expect(closeButton).not.toBeNull();

    if (closeButton) {
      fireEvent.click(closeButton);
    }

    expect(mockOnCloseHandler).toHaveBeenCalledTimes(1);

    expect(closeButton).toMatchSnapshot();
  });
});
