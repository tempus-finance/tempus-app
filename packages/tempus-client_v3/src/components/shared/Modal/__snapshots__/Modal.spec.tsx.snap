// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal calls onClose when user clicks on backdrop 1`] = `
<div>
  <div
    class="tc__modal-backdrop"
    id="modal-backdrop"
  />
  <div
    class="tc__modal-container "
  >
    <div
      class="tc__modal tc__modal__plain tc__modal__size-small"
    >
      <div
        class="tc__modal-header"
      >
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        />
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <svg
            class="tc__icon tc__icon-close"
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
              fill="#222222"
            />
          </svg>
        </button>
      </div>
      <div
        class="tc__modal__body"
      >
        This is modal content.
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal calls onClose when user clicks on the close button 1`] = `
<button
  class="tc__btn "
  data-selected="false"
  type="button"
>
  <svg
    class="tc__icon tc__icon-close"
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
      fill="#222222"
    />
  </svg>
</button>
`;

exports[`Modal does not render child elements when modal is closed 1`] = `<div />`;

exports[`Modal renders child elements when large modal is open 1`] = `
<div>
  <div
    class="tc__modal-backdrop"
    id="modal-backdrop"
  />
  <div
    class="tc__modal-container "
  >
    <div
      class="tc__modal tc__modal__plain tc__modal__size-large"
    >
      <div
        class="tc__modal-header"
      >
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <svg
            class="tc__icon tc__icon-close"
            fill="none"
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
              fill="#222222"
            />
          </svg>
        </button>
      </div>
      <div
        class="tc__modal__body"
      >
        
        This is modal content.
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders child elements when small modal is open 1`] = `
<div>
  <div
    class="tc__modal-backdrop"
    id="modal-backdrop"
  />
  <div
    class="tc__modal-container "
  >
    <div
      class="tc__modal tc__modal__plain tc__modal__size-small"
    >
      <div
        class="tc__modal-header"
      >
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        />
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <svg
            class="tc__icon tc__icon-close"
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
              fill="#222222"
            />
          </svg>
        </button>
      </div>
      <div
        class="tc__modal__body"
      >
        This is modal content.
      </div>
    </div>
  </div>
</div>
`;
